src/Data/Aeson/Types/FromJSON.hs

```diff
@@ -2352,67 +2382,45 @@ instance FromJSON a => FromJSON (Monoid.Last a) where
     omittedField = omittedField1
 
 instance FromJSON1 Semigroup.Min where
-    liftParseJSON _ p _ a = coerce (p a)
-
-    liftParseJSONList _ _ p a = coerce (p a)
+    liftParseJSON _ p _ = coerce p
 
     liftOmittedField = coerce
 
 instance (FromJSON a) => FromJSON (Semigroup.Min a) where
     parseJSON = parseJSON1
 
-    parseJSONList = liftParseJSONList omittedField parseJSON parseJSONList
-
-    omittedField = omittedField1
-
 instance FromJSON1 Semigroup.Max where
-    liftParseJSON _ p _ a = coerce (p a)
+    liftParseJSON _ p _ = coerce p
 
-    liftParseJSONList _ _ p a = coerce (p a)
     liftOmittedField = coerce
 
 instance (FromJSON a) => FromJSON (Semigroup.Max a) where
     parseJSON = parseJSON1
 
-    parseJSONList = liftParseJSONList omittedField parseJSON parseJSONList
-    omittedField = omittedField1
-
 instance FromJSON1 Semigroup.First where
-    liftParseJSON _ p _ a = coerce (p a)
+    liftParseJSON _ p _ = coerce p
 
-    liftParseJSONList _ _ p a = coerce (p a)
     liftOmittedField = coerce
 
 instance (FromJSON a) => FromJSON (Semigroup.First a) where
     parseJSON = parseJSON1
 
-    parseJSONList = liftParseJSONList omittedField parseJSON parseJSONList
-
-
 instance FromJSON1 Semigroup.Last where
-    liftParseJSON _ p _ a = coerce (p a)
+    liftParseJSON _ p _ = coerce p
 
-    liftParseJSONList _ _ p a = coerce (p a)
     liftOmittedField = coerce
 
 instance (FromJSON a) => FromJSON (Semigroup.Last a) where
     parseJSON = parseJSON1
 
-    parseJSONList = liftParseJSONList omittedField parseJSON parseJSONList
-    omittedField = omittedField1
-
 instance FromJSON1 Semigroup.WrappedMonoid where
-    liftParseJSON _ p _ a = coerce (p a)
+    liftParseJSON _ p _ = coerce p
 
-    liftParseJSONList _ _ p a = coerce (p a)
```

Comment:
this removals are wrong.

Reply:
I added it back, and made it coherent with the rest of the code

Reply:
I also wonder whether those two lines : 

```
parseJSONList = liftParseJSONList omittedField parseJSON parseJSONList
omittedField = omittedField1
```

should be added to all the FromJSON instances from Data.Monoid. If you think so I can do this.